From e1d05a6c77ca2f206d53980f5d3de3958e86a3d7 Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Sun, 3 Jan 2021 17:54:47 +0100
Subject: [PATCH 2/2] Provide option to use pre-downloaded archive

Live-download is problematic when used in packaging systems, since
those systems have their own download caches. Thus, provide an
option to allow separately downloading the model data and then
reusing it in the build.
---
 CMakeLists.txt      |  1 +
 data/CMakeLists.txt | 42 +++++++++++++++++++++++++++++-------------
 2 files changed, 30 insertions(+), 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index afb4249..467c90c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,6 +68,7 @@ find_package(GLIB2 REQUIRED)
 # Options
 
 option(INSTALL_UTILS "Install generator utilities" OFF)
+option(DOWNLOAD_MODEL_DATA_ARCHIVE "Update the model data archive while building" ON)
 
 # DBM: BerkeleyDB
 find_package(BerkeleyDB)
diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index d3df1d0..e5422ff 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -40,19 +40,35 @@ add_custom_target(
         ${BINARY_MODEL_DATA}
 )
 
-add_custom_command(
-    OUTPUT
-        ${CMAKE_SOURCE_DIR}/data/gb_char.table
-        ${CMAKE_SOURCE_DIR}/data/gbk_char.table
-        ${CMAKE_SOURCE_DIR}/data/interpolation2.text
-        ${CMAKE_SOURCE_DIR}/data/table.conf
-    COMMENT
-        "Downloading textual model data..."
-    COMMAND
-       wget http://downloads.sourceforge.net/libpinyin/models/model19.text.tar.gz
-    COMMAND
-       tar xvf model19.text.tar.gz -C ${CMAKE_SOURCE_DIR}/data
-)
+if(DOWNLOAD_MODEL_DATA_ARCHIVE)
+    add_custom_command(
+        OUTPUT
+            ${CMAKE_SOURCE_DIR}/data/gb_char.table
+            ${CMAKE_SOURCE_DIR}/data/gbk_char.table
+            ${CMAKE_SOURCE_DIR}/data/interpolation2.text
+            ${CMAKE_SOURCE_DIR}/data/table.conf
+        COMMENT
+            "Downloading textual model data..."
+        COMMAND
+        wget -P ${CMAKE_SOURCE_DIR} http://downloads.sourceforge.net/libpinyin/models/model19.text.tar.gz
+        COMMAND
+        tar xvf ${CMAKE_SOURCE_DIR}/model19.text.tar.gz -C ${CMAKE_SOURCE_DIR}/data
+    )
+else()
+    add_custom_command(
+        OUTPUT
+            ${CMAKE_SOURCE_DIR}/data/gb_char.table
+            ${CMAKE_SOURCE_DIR}/data/gbk_char.table
+            ${CMAKE_SOURCE_DIR}/data/interpolation2.text
+            ${CMAKE_SOURCE_DIR}/data/table.conf
+        COMMENT
+            "Checking if model data is provided in ${CMAKE_SOURCE_DIR}/model19.text.tar.gz"
+        COMMAND
+        test -f ${CMAKE_SOURCE_DIR}/model19.text.tar.gz
+        COMMAND
+        tar xvf ${CMAKE_SOURCE_DIR}/model19.text.tar.gz -C ${CMAKE_SOURCE_DIR}/data
+    )
+endif()
 
 add_custom_command(
     OUTPUT
-- 
2.27.0

